c/openbsd-6.2/Makefile

```diff
@@ -0,0 +1,46 @@
+CC.Arch := 64
+
+CFLAGS=-g -Werror -Wall -Wimplicit-function-declaration -Wno-uninitialized \
+-Wno-pointer-sign -Wno-address-of-packed-member -Wno-constant-conversion \
+-Wframe-larger-than=2047 -mcmodel=kernel -mno-red-zone -mno-sse2 -mno-sse \
+-mno-3dnow -mno-mmx -msoft-float -fno-omit-frame-pointer -ffreestanding \
+-fno-pie -O2 -pipe -nostdinc -I${CURDIR}/sources/sys -DDIAGNOSTIC -DKTRACE -DACCOUNTING \
+-DKMEMSTATS -DPTRACE -DPOOL_DEBUG -DCRYPTO -DSYSVMSG -DSYSVSEM -DSYSVSHM \
+-DUVM_SWAP_ENCRYPT -DFFS -DFFS2 -DFFS_SOFTUPDATES -DUFS_DIRHASH -DQUOTA \
+-DEXT2FS -DMFS -DNFSCLIENT -DNFSSERVER -DCD9660 -DUDF -DMSDOSFS -DFIFO \
+-DFUSE -DSOCKET_SPLICE -DTCP_ECN -DINET6 -DPPP_BSDCOMP -DPPP_DEFLATE \
+-DPIPEX -DMROUTING -DBOOT_CONFIG -DUSER_PCICONF -DAPERTURE -DMTRR -DNTFS \
+-DHIBERNATE -DPCIVERBOSE -DUSBVERBOSE -DWSDISPLAY_COMPAT_USL \
+-DWSDISPLAY_COMPAT_RAWKBD -DWSDISPLAY_DEFAULTSCREENS="6" -DX86EMU \
+-DONEWIREVERBOSE -DMAXUSERS=80 -D_KERNEL -c -E -P
+
+.PHONY: all %.i
+
+all: if_etherip-fixed-double-free.i if_etherip-double-free.i
+	[ -f ${CURDIR}/sources/sys/if_etherip.c.orig ] && \
+	mv ${CURDIR}/sources/sys/if_etherip.c.orig \
+	${CURDIR}/sources/sys/if_etherip.c || echo "No .orig file exists."
+
+%.i: %.c
```

Comment:
What do these rules do, what are there intention?

Usually, these make files are just for checking that the files are compilable, everything else should be documented and explained.